From 88e90ee4c36f0decdf7c15e271bccb8b0809a21a Mon Sep 17 00:00:00 2001
From: Stefan Strogin <stefan@steils.org>
Date: Tue, 24 Nov 2020 11:08:13 +0200
Subject: [PATCH] trousers: do not re-declare RSA_set0_key with LibreSSL

The method RSA_set0_key is provided by LibreSSL since 2.7.0, that means
in all currently supported versions.

Upstream-Status: Submitted
[https://sourceforge.net/p/trousers/trousers/merge-requests/2/]
Signed-off-by: Stefan Strogin <stefan@steils.org>
---
 src/trspi/crypto/openssl/rsa.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/trspi/crypto/openssl/rsa.c b/src/trspi/crypto/openssl/rsa.c
index 24f13e7..cf46c08 100644
--- a/src/trspi/crypto/openssl/rsa.c
+++ b/src/trspi/crypto/openssl/rsa.c
@@ -38,7 +38,7 @@
 #define DEBUG_print_openssl_errors()
 #endif
 
-#if (OPENSSL_VERSION_NUMBER < 0x10100001L) || defined(LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER < 0x10100001L)
 static int
 RSA_set0_key(RSA *r, BIGNUM *n, BIGNUM *e, BIGNUM *d)
 {
-- 
2.29.2

